qlty-check/src/executor.rs

```diff
@@ -245,223 +246,169 @@ impl Executor {
+        info!(
+            "Staged {} workspace entries and {} config files in {:.2}s",
+            self.plan.workspace_entry_paths().len(),
+            loaded_config_files.len(),
+            timer.elapsed().as_secs_f32()
+        );
 
-        // load exported config paths before anything else
-        for config_file in &exported_config_paths {
-            if self.plan.workspace.root != self.plan.staging_area.destination_directory {
-                // for formatters
-                let loaded_config_file = load_config_file_from_source(
-                    config_file,
-                    &self.plan.staging_area.destination_directory,
-                )?;
+        Ok(loaded_config_files)
+    }
 
-                if !loaded_config_file.is_empty() {
-                    loaded_config_files.push(loaded_config_file);
-                }
-            }
+    fn execute_config_staging_operations(&self) -> Result<Vec<String>> {
+        let mut loaded_config_files = Vec::new();
+        let mut download_cache: HashMap<String, Vec<u8>> = HashMap::new();
 
-            // for linters
-            let loaded_config_file =
-                load_config_file_from_source(config_file, &self.plan.workspace.root)?;
+        for operation in &self.plan.config_staging_operations {
+            let maybe_loaded = self
+                .execute_single_config_operation(operation, &mut download_cache)
+                .with_context(|| {
+                    format!("Failed to execute config staging operation: {operation:?}")
+                })?;
 
-            if !loaded_config_file.is_empty() {
-                loaded_config_files.push(loaded_config_file);
+            if let Some(loaded_file) = maybe_loaded {
+                loaded_config_files.push(loaded_file);
             }
         }
 
-        self.check_and_copy_configs_into_tool_install(&mut loaded_config_files)?;
-        self.plan_plugins_fetch(&mut loaded_config_files)?;
-
-        for config_file in &repository_config_files {
-            if let Err(err) = load_config_file_from_repository(
-                config_file,
-                &self.plan.workspace,
-                &self.plan.staging_area.destination_directory,
-            ) {
-                error!("Failed to load config file from repository: {:?}", err);
-            }
-        }
+        Ok(loaded_config_files)
+    }
 
-        for config_file in &config_file_names {
-            if self.plan.workspace.root != self.plan.staging_area.destination_directory {
-                // for formatters
-                let loaded_config_file = load_config_file_from_qlty_dir(
-                    config_file,
-                    &self.plan.workspace,
-                    &self.plan.staging_area.destination_directory,
-                )?;
-
-                if !loaded_config_file.is_empty() {
-                    loaded_config_files.push(loaded_config_file);
-                }
+    fn execute_single_config_operation(
```

Comment:
Starting here we have the code to execute the ConfigOperations, which splits based on the config source (download or file). This seems this code would be better off not being in `executor.rs` which is fairly big.

Config file are an important concept. In the planner, we have `planner/config_files.rs` which makes sense to me. I would recommend we move the code for executing config operations to `executor/config_files.rs`.

This creates a _symmetry_ between the planner and executor.